zerver/lib/bugdown/fenced_code.py

```diff
@@ -107,6 +108,24 @@
 CODE_WRAP = '<pre><code%s>%s\n</code></pre>'
 LANG_TAG = ' class="%s"'
 
+def validate_curl_content(lines: List[str]) -> None:
+    error_msg = """
+Missing required -X argument in curl command:
+
+{command}
+""".strip()
+
+    for line in lines:
+        regex = r"curl \-X \"?(GET|DELETE|PATCH|POST)\"?"
+        if line.startswith('curl'):
+            if re.search(regex, line) is None:
+                raise BugdownRenderingException(error_msg.format(command=line.strip()))
+
+
+CODE_VALIDATORS = {
+    'curl': validate_curl_content,
+}
```

Comment:
I think whether this happens needs to be an option passed into the markdown processor.  Otherwise, this behavior change will impact not just the API docs but also Zulip's message formatting, which we really don't want.  So 2 requests:
* Make this an option we pass into `zerver.lib.bugdown.fenced_code.makeExtension` in the `app_filters.py` code.
* Add a test to `test_bugdown.py` that this exception doesn't fire for curl-formatted code.

Reply:
What do you mean by "curl-formatted" code?